app/controllers/application_controller.rb

```diff
@@ -26,6 +27,25 @@ def authorize(record, query=nil, user=nil)
     end
   end
 
+  def param_date(sym)
+    begin
+      Date.strptime(params[sym].to_s)
+    rescue
+      nil
+    end
```

Comment:
Mind checking for the possible error scenarios, rather than a catch-all `rescue`?